modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -2708,10 +2728,21 @@ private Expression<?>[] _getSelectExpressions(
 									getObjectDefinitionId()),
 							relatedObjectDefinition.getExtensionDBTableName());
 
+				if (objectRelationship.isSelf()) {
+					relatedDynamicObjectDefinitionTable =
+						relatedDynamicObjectDefinitionTable.as(
+							"dynamicObjectDefinitionTableSub");
+					relatedExtensionDynamicObjectDefinitionTable =
+						relatedExtensionDynamicObjectDefinitionTable.as(
+							"extensionDynamicObjectDefinitionTableSub");
+				}
+
 				JoinStep joinStep = DSLQueryFactoryUtil.select(
 					_getFunctionExpression(
 						objectFieldSettingsValues, relatedObjectDefinition,
-						relatedDynamicObjectDefinitionTable)
+						relatedDynamicObjectDefinitionTable,
+						relatedExtensionDynamicObjectDefinitionTable,
```

Comment:
Yes, I think it's better that way too.

Reply:
I don't think so, because currently there's a limitation that you can't put aliases in the conditions coming from the filters. They are created [here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java#L2811C42-L2811C42).

Reply:
@pedro-oliveira446 but the `_getFunctionExpression` is only using the `DynamicObjectDefinitionTable` to get the `PrimaryKeyColumn` [here](https://github.com/liferay/liferay-portal/blob/19202b7d511d699e8e2453ba7725964ded3ad2ad/modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java#L2100C13-L2100C48).

Would this affect the code related to the filter?

Reply:
As we can see [here](https://github.com/liferay/liferay-portal/blob/19202b7d511d699e8e2453ba7725964ded3ad2ad/modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java#L2100C13-L2100C48), the creation of the filter does not use the dynamics tables that I have aliased, which generates an error in the query.